### PR TITLE
Applied correct reshaping to metric func sparse_top_k

### PR DESCRIPTION
**Pull request by dfossl**
For issue [#15939](https://github.com/keras-team/keras/issues/15939).

Simple fix, was calling tf.math.in_top_k in the background but needed to reshape the inputs to make the call and was failing to reshape the results upon return. I just reshaped the results before returning to match the input. This gives the expected behaviour and allows sample_weights to be applied without issue.

```
target = tf.constant([[1,1,2,0,0,0],[1,1,2,2,0,0]])
predict = tf.constant([[[0,1,0],
                       [0,1,0],
                       [0,0,1],
                       [0,1,0],
                       [1,0,0],
                       [1,0,0]],
                       [[0,1,0],
                        [0,1,0],
                        [0,0,1],
                        [0,0,1],
                        [0,1,0],
                        [0,1,1]]], dtype=tf.float32)

mask = tf.math.logical_not(tf.math.equal(target, 0))

mask = tf.cast(mask, dtype=tf.int64)

m = tf.keras.metrics.SparseTopKCategoricalAccuracy(k=1)
m.update_state(target, predict, sample_weight=mask)
m.result().numpy()
```
Gives:
```
1
```
No errors.

To be safe I ran all tests and they passed. There isn't a test written for this specific functionality but considering it's pretty straight forward I wasn't sure if new tests were necessary. I tested on my original issue code to see if it passed and that worked fine, and I ran it on all the documentation examples for the function and they also worked as expected. If tests are needed let me know.

### CODE REVIEW

**Comment by gbaned** [comment]
@dfossl  Can you please resolve conflicts? Thanks!

**Comment by dfossl** [comment]
> @dfossl Can you please resolve conflicts? Thanks!

@gbaned Hey, not sure I can? I just see the "This branch has conflicts that must be resolved" for keras/metric.py but I can't see any of the conflicts or press the "Resolve conflicts" button. Also, the conflicts are in the only file I changed so wouldn't it just be my changes? Not sure I have access to do anything else.
This is my first keras contribution and I am new to it so if I have something wrong in setting up the PR let me know👍

**Comment by dfossl** [comment]
@gbaned  I think conflicts should be resolved now? Seems the only conflict was the directory structure change. Let me know if there is a mistake I made.

**Review by fchollet — APPROVED** [review]
Very nice, thank you for the PR!

I'll make a note to increase `sample_weight` unit testing coverage for metrics.
